File: sdk/typescript/src/api/rating-plans.ts
import { HTTPClient } from './http-client';
import { RatingPlan } from '../types';

export class RatingPlanAPI {
  constructor(private client: HTTPClient) {}

  async list(): Promise<RatingPlan[]> {
    return this.client.request<RatingPlan[]>({
      method: 'GET',
      endpoint: '/v1/rating-plans',
    });
  }

  async get(planId: string): Promise<RatingPlan> {
    return this.client.request<RatingPlan>({
      method: 'GET',
      endpoint: `/v1/rating-plans/${planId}`,
    });
  }
}

File: sdk/rust/src/api/rating_plans.rs
use crate::client::HTTPClient;
use crate::error::TelecomError;
use crate::types::RatingPlan;

pub struct RatingPlanAPI {
    client: HTTPClient,
}

impl RatingPlanAPI {
    pub fn new(client: HTTPClient) -> Self {
        Self { client }
    }

    pub async fn list(&self) -> Result<Vec<RatingPlan>, TelecomError> {
        self.client.get("/v1/rating-plans", None).await
    }

    pub async fn get(&self, plan_id: &str) -> Result<RatingPlan, TelecomError> {
        self.client
            .get(&format!("/v1/rating-plans/{}", plan_id), None)
            .await
    }
}
